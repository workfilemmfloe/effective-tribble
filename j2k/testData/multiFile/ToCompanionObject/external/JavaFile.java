package test;

class C {
    void foo(ClassWithStatics c) {
        ClassWithStatics.staticMethod(ClassWithStatics.staticField);
        c.instanceMethod();
        ClassWithStatics.staticField += 2;
    }
}

class D extends ClassWithStatics {
    void foo() {
        staticMethod(staticField);
        ourValue *= 2;
    }
}