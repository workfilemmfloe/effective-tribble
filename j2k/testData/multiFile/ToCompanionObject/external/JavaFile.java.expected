package test;

class C {
    void foo(ClassWithStatics c) {
        ClassWithStatics.Companion.staticMethod(ClassWithStatics.staticField);
        c.instanceMethod();
        ClassWithStatics.staticField += 2;
    }

    void methodReferences() {
        JFunction1ReturnType<Integer> staticMethod = ClassWithStatics.Companion::staticMethod;
        JFunction1ReturnType<ClassWithStatics> instanceMethod = ClassWithStatics::instanceMethod;
    }
}

class D extends ClassWithStatics {
    void foo() {
        Companion.staticMethod(staticField);
        value *= 2;
    }
}